projects/angular/src/data/tree-view/tree-focus-manager.service.ts

```diff
@@ -99,6 +99,71 @@ export class TreeFocusManagerService<T> {
     }
   }
 
+  private findNodeStartsWith(searchString: string, model: TreeNodeModel<T>): TreeNodeModel<T> {
+    if (!model) {
+      return null;
+    }
+
+    if (model.textContent.startsWith(searchString)) {
+      return model;
+    }
+
+    if (model.expanded && model.children.length > 0) {
+      for (const childModel of model.children) {
```

Comment:
The code pattern from lines 111-118 seems repeated in this file several times. This seems like a good opportunity to abstract that code pattern and dry up this code...